specification/confidentialledger/resource-manager/Microsoft.ConfidentialLedger/preview/2023-06-28-preview/confidentialledger.json

```diff
@@ -0,0 +1,830 @@
+        "ledgerUri": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Endpoint for calling Ledger Service."
+        },
+        "identityServiceUri": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Endpoint for accessing network identity."
+        },
+        "ledgerInternalNamespace": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Internal namespace for the Ledger"
+        },
+        "runningState": {
+          "description": "Object representing RunningState for Ledger.",
+          "$ref": "#/definitions/RunningState"
+        },
+        "ledgerType": {
+          "description": "Type of Confidential Ledger",
+          "$ref": "#/definitions/ConfidentialLedgerType"
+        },
+        "provisioningState": {
+          "description": "Provisioning state of Ledger Resource",
+          "readOnly": true,
+          "$ref": "#/definitions/ProvisioningState"
+        },
+        "aadBasedSecurityPrincipals": {
+          "description": "Array of all AAD based Security Principals.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AADBasedSecurityPrincipal"
+          },
+          "x-ms-identifiers": [
+            "principalId"
+          ]
+        },
+        "certBasedSecurityPrincipals": {
+          "description": "Array of all cert based Security Principals.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CertBasedSecurityPrincipal"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "ConfidentialLedgerBackup": {
+      "description": "Object representing Backup properties of Confidential Ledger Resource.",
+      "type": "object",
+      "properties": {
+        "restoreRegion": {
+          "type": "string",
+          "readOnly": false,
+          "description": "The region where the backup of the Ledger Resource will eventually be restored to."
+        },
+        "uri": {
+          "type": "string",
+          "readOnly": false,
+          "description": "SAS URI used to access the backup Fileshare."
+        }
+      }
+    },
+    "ConfidentialLedgerRestore": {
+      "description": "Object representing Restore properties of Confidential Ledger Resource.",
+      "type": "object",
+      "properties": {
+        "fileShareName": {
+          "type": "string",
+          "readOnly": false,
+          "description": "Fileshare Name where the Confidential Ledger Resource backup is stored."
+        },
+        "restoreRegion": {
+          "type": "string",
+          "readOnly": false,
+          "description": "The region the Ledger Resource is being restored to."
+        },
+        "uri": {
+          "type": "string",
+          "readOnly": false,
+          "description": "SAS URI used to access the backup Fileshare."
+        }
+      }
+    },
+    "ConfidentialLedgerBackupRestoreResponse": {
```

Comment:
future-proofing - consider defining separate response payloads for each action in case you want to diverge them in later api-versions

Reply:
Yes, good point! I separated them.